designer/src/components/TemplateBuilder.tsx

```diff
@@ -0,0 +1,763 @@
+        return {
+          id: Math.random().toString(36).substring(2, 9),
+          type: 'if',
+          content: content,
+          children: rest[2].map(convertMustacheTokenToBlock).filter(Boolean),
+        };
+
+      case '^':
+        return {
+          id: Math.random().toString(36).substring(2, 9),
+          type: 'unless',
+          content: content,
+          children: rest[2].map(convertMustacheTokenToBlock).filter(Boolean),
+        };
+
+      default:
+        return null;
+    }
+  } catch (error) {
+    console.warn('Error converting token:', error);
+    return null;
+  }
+};
+
+/**
+ * Type guard for TemplateBlock
+ */
+function isTemplateBlock(block: TemplateBlock | null): block is TemplateBlock {
+  return block !== null;
+}
+
+/**
+ * Component for editing a single template block
+ */
+const TemplateBlockEditor: React.FC<{
+  block: TemplateBlock;
+  variables: Variable[];
+  systemVariables: Variable[];
+  onUpdate: (block: TemplateBlock) => void;
+  onDelete: () => void;
+  onMoveUp?: () => void;
+  onMoveDown?: () => void;
+  isFirst?: boolean;
+  isLast?: boolean;
+  level?: number;
+}> = ({
+  block,
+  variables,
+  systemVariables,
+  onUpdate,
+  onDelete,
+  onMoveUp,
+  onMoveDown,
+  isFirst,
+  isLast,
+  level = 0,
+}) => {
+  const allVariables = [...variables, ...systemVariables];
+
+  const updateBlockContent = (content: string) => {
+    onUpdate({...block, content});
+  };
+
+  const addChildBlock = () => {
+    const newBlock: TemplateBlock = {
+      id: Math.random().toString(36).substring(2, 9),
+      type: 'text',
+      content: '',
+    };
+    onUpdate({
+      ...block,
+      children: [...(block.children || []), newBlock],
+    });
+  };
+
+  return (
+    <Box sx={{pl: 3, borderLeft: '2px solid', borderColor: 'divider', mb: 2}}>
+      <Grid container spacing={2} alignItems="center">
+        <Grid item xs={12} sm="auto">
+          <FormControl fullWidth sx={{minWidth: 120}}>
+            <InputLabel>Block Type</InputLabel>
+            <Select
+              value={block.type}
+              label="Block Type"
+              onChange={e =>
+                onUpdate({...block, type: e.target.value as BlockType})
+              }
+            >
+              <MenuItem value="text">Text</MenuItem>
+              <MenuItem value="variable">Variable</MenuItem>
+              <MenuItem value="if">Conditional</MenuItem>
+              <MenuItem value="unless">Unless</MenuItem>
+            </Select>
+          </FormControl>
+        </Grid>
+
+        {block.type === 'variable' && (
+          <Grid item xs={12} sm>
+            <FormControl fullWidth>
+              <InputLabel>Variable</InputLabel>
+              <Select
+                value={block.content}
+                label="Variable"
+                onChange={e => updateBlockContent(e.target.value)}
+              >
+                <MenuItem value="">Choose variable...</MenuItem>
+                {allVariables.map(v => (
```

Comment:
You should only be able to add fields from this form to the expression. Doesn't make sense to add a field from another form since it can't have a value here.